test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/FreezeUnfreezeTokenPrecompileSuite.java

```diff
@@ -95,7 +93,136 @@ public List<HapiApiSpec> getSpecsInSuite() {
+                                .exposingCreatedIdTo(accountID::set),
+                        cryptoCreate(TOKEN_TREASURY),
+                        tokenCreate(VANILLA_TOKEN)
+                                .tokenType(FUNGIBLE_COMMON)
+                                .treasury(TOKEN_TREASURY)
+                                .freezeKey(FREEZE_KEY)
+                                .initialSupply(1_000)
+                                .exposingCreatedIdTo(id -> vanillaTokenID.set(asToken(id))),
+                        uploadInitCode(FREEZE_CONTRACT),
+                        contractCreate(FREEZE_CONTRACT),
+                        tokenAssociate(ACCOUNT, VANILLA_TOKEN))
+                .when(
+                        withOpContext(
+                                (spec, opLog) ->
+                                        allRunFor(
+                                                spec,
+                                                contractCall(
+                                                                FREEZE_CONTRACT,
+                                                                TOKEN_UNFREEZE_FUNC,
+                                                                notAnAddress,
+                                                                asAddress(accountID.get()))
+                                                        .payingWith(ACCOUNT)
+                                                        .gas(GAS_TO_OFFER)
+                                                        .via("UnfreezeTx")
+                                                        .hasKnownStatus(CONTRACT_REVERT_EXECUTED),
+                                                cryptoUpdate(ACCOUNT).key(FREEZE_KEY),
+                                                contractCall(
+                                                                FREEZE_CONTRACT,
+                                                                TOKEN_FREEZE_FUNC,
+                                                                notAnAddress,
+                                                                asAddress(accountID.get()))
+                                                        .hasKnownStatus(CONTRACT_REVERT_EXECUTED)
+                                                        .payingWith(ACCOUNT)
+                                                        .gas(GAS_TO_OFFER)
+                                                        .via("FreezeTx"))))
+                .then(
+                        childRecordsCheck(
+                                "UnfreezeTx",
+                                CONTRACT_REVERT_EXECUTED,
+                                recordWith().status(INVALID_TOKEN_ID)),
+                        childRecordsCheck(
+                                "FreezeTx",
+                                CONTRACT_REVERT_EXECUTED,
+                                recordWith().status(INVALID_TOKEN_ID)));
+    }
+
+    private HapiApiSpec noKeyReverts() {
```

Comment:
There's already a case for a token without a freeze key calling the freeze function in "freezeUnfreezeFungibleWithNegativeCases".  Maybe you can add the unfreeze function call there also, this will eliminate the need for the "noKeyReverts" test case.